.github/workflows/on-push-do-doco.yml

```diff
@@ -0,0 +1,28 @@
+name: on-push-do-doco
+on:
+  # Only update the docs on default branch, to reduce the chance of conflicts
+  push:
+    branches:
+      - master
+jobs:
+  release:
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run MarkdownSnippets
+      run: |
+        dotnet tool install --global MarkdownSnippets.Tool
+        mdsnippets ${GITHUB_WORKSPACE} \
+          --convention InPlaceOverwrite \
+          --exclude-directories 'cmake-build-*' \
+          --toc-level 5
+      shell: bash
+    - name: Push changes
+      run: |
+        git config --local user.email "action@github.com"
+        git config --local user.name "GitHub Action"
+        git commit -m "[docs] Regenerate tables of contents" -a  || echo "nothing to commit"
+        remote="https://${GITHUB_ACTOR}:${{secrets.GITHUB_TOKEN}}@github.com/${GITHUB_REPOSITORY}.git"
+        branch="${GITHUB_REF:11}"
```

Comment:
I'm having a look and the best I've found so far have been this SO question (https://stackoverflow.com/questions/58033366/how-to-get-current-branch-within-github-actions). I can't believe there is no easier way to get the branch name 😅 

Maybe `${GITHUB_REF#refs/heads/}` is more clear.

Reply:
I've just run it on Mac, and it does... (the mdsnippets tool is available on Mac, Linux and Windows - the CI job uses windows as that's what the author users...


```mdsnippets --convention InPlaceOverwrite --exclude-directories 'recipes' --toc-level 5
Config:
    TargetDirectory: /Users/clare/Documents/develop/Conan/conan-center-index-claremacrae
    UrlPrefix: 
    LinkFormat: GitHub
    Convention: InPlaceOverwrite
    TocLevel: 5
    ValidateContent: False
    HashSnippetAnchors: False
    TreatMissingAsWarning: False
    FileConfigPath:  (exists:False)
    ExcludeDirectories:
        recipes
Found 4 files for snippets
MarkdownProcessor Finished. 49ms
Finished 283ms
```

I know from past runs that there's no way that this would have taken only 283 ms if it had traversed recipes...

Reply:
I'm unsure if you need any input from me at this stage on the GITHUB_REF comments... 

I just took the YAML code straight from the author of the mdsnippets tool: 

https://github.com/SimonCropp/MarkdownSnippets/blob/master/docs/on-push-do-docs.yml

Reply:
On second thoughts, I did some experimenting - using info from https://stackoverflow.com/a/59780579/104370

Which pointed to this:
https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context

In this run.... https://github.com/claremacrae/commandline-videostore-cpp/runs/1922585010?check_suite_focus=true

I had this step:

```yml
      - name: Check Branch references
        run: |
          echo "github.ref = ${{ github.ref }}"
          echo "github.head_ref = ${{ github.head_ref }}"
          echo "github.base_ref = ${{ github.base_ref }}"
          echo "GITHUB_REF = $GITHUB_REF"
          echo "GITHUB_HEAD_REF = $GITHUB_HEAD_REF"
          echo "GITHUB_BASE_REF = $GITHUB_BASE_REF"
```

And the output was this - where `starting-point` was the name of the branch:

```
Run echo "github.ref = refs/heads/starting-point"
  echo "github.ref = refs/heads/starting-point"
  echo "github.head_ref = "
  echo "github.base_ref = "
  echo "GITHUB_REF = $GITHUB_REF"
  echo "GITHUB_HEAD_REF = $GITHUB_HEAD_REF"
  echo "GITHUB_BASE_REF = $GITHUB_BASE_REF"
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
github.ref = refs/heads/starting-point
github.head_ref = 
github.base_ref = 
GITHUB_REF = 
GITHUB_HEAD_REF = 
GITHUB_BASE_REF = 
```

So `${{ github.ref }}` looks like the right answer to me...

This is based on the belief that you don't have to worry about references relating to pull requests, as the script deliberately only acts on push to master:

```yml
on:
  # Only update the docs on default branch, to reduce the chance of conflicts
  push:
    branches:
      - master
```